dashboard/lib/school_info_deduplicator.rb

```diff
@@ -0,0 +1,45 @@
+module SchoolInfoDeduplicator
+  # Returns true if the SchoolInfo already exists and we should reuse that.
+  # Returns false if the SchoolInfo is new and should be stored.
+  # Validates the SchoolInfo first so that we fall into the latter path in
+  # that case.
+  def deduplicate_school_info(school_info_attr, self_object)
```

Comment:
Also the method name is misleading. It doesn't actually dedupe, but rather it answers the question of whether these attributes represent a duplicate entry. Rename something like `duplicate_school_info?(...)`